framework/src/base/MeshGeneratorSystem.C

```diff
@@ -0,0 +1,494 @@
+                     " '",
+                     (*find_sub_dependency)->name(),
+                     "', but was not declared as a sub dependency.\n\nTo correct this, declare the "
+                     "mesh in the parameter '",
+                     param_name,
+                     "' as a sub dependency with declareMeshForSub().");
+    // Isn't used at all
+    else
+      mg->mooseError(
+          "You failed to request the generated mesh(es) for the parameter '",
+          param_name,
+          "'.\n\nIn specific, the mesh from MeshGenerator '",
+          dependency_name,
+          "' was not requested.\n\nTo correct this, you should remove the parameter if the "
+          "mesh(es)\nare not needed, or request the mesh(es) with getMesh()/getMeshes().");
+  }
+
+  mooseAssert(!_mesh_generators.count(generator_name), "Already created");
+  _mesh_generators.emplace(generator_name, mg);
+  mooseAssert(!_mesh_generator_outputs.count(generator_name), "Already exists");
+  _mesh_generator_outputs[generator_name];
+  _mesh_generator_params.erase(find_params);
+
+  return mg;
+}
+
+std::unique_ptr<MeshBase> &
+MeshGeneratorSystem::getMeshGeneratorOutput(const MeshGeneratorName & name)
+{
+  mooseAssert(_app.constructingMeshGenerators() ||
+                  _app.actionWarehouse().getCurrentTaskName() == "execute_mesh_generators",
+              "Incorrect call time");
+
+  auto it = _mesh_generator_outputs.find(name);
+  mooseAssert(it != _mesh_generator_outputs.end(), "Not initialized");
+  it->second.push_back(nullptr);
+  return it->second.back();
+}
+
+bool
+MeshGeneratorSystem::hasMeshGenerator(const MeshGeneratorName & name) const
+{
+  return _mesh_generators.count(name);
+}
+
+bool
+MeshGeneratorSystem::hasMeshGeneratorParams(const MeshGeneratorName & name) const
+{
+  return _mesh_generator_params.count(name);
+}
+
+const MeshGenerator &
+MeshGeneratorSystem::getMeshGenerator(const std::string & name) const
+{
+  const auto it = _mesh_generators.find(name);
+  if (it == _mesh_generators.end())
+    mooseError("Failed to find a MeshGenerator with the name '", name, "'");
+  mooseAssert(it->second, "Invalid shared pointer");
+  return *it->second;
+}
+
+std::vector<std::string>
+MeshGeneratorSystem::getMeshGeneratorNames() const
+{
+  std::vector<std::string> names;
+  for (auto & pair : _mesh_generators)
+    names.push_back(pair.first);
+  return names;
+}
+
+std::unique_ptr<MeshBase>
+MeshGeneratorSystem::getMeshGeneratorMesh(const bool check_unique /* = true */)
+{
+  if (_popped_final_mesh_generator == true)
+    mooseError("MeshGeneratorSystem::getMeshGeneratorMesh is being called for a second time. You "
+               "cannot do "
+               "this because the final generated mesh was popped from its storage container the "
+               "first time this method was called");
+
+  if (_final_generated_meshes.empty())
+    mooseError("No generated mesh to retrieve. Your input file should contain either a [Mesh] or "
```

Comment:
missing something here. 
not sure what you mean by "block"

Reply:
This is old code. Shouldn't hit this unless the SetupMeshAction fails, so made it a developer error